### DISCUSSION TITLE
Диалоги с Драконами

### DISCUSSION THREAD

**Discussion by Kor-Angar** [dosinabox/g2nr_unofficial_update#511] [Show and tell]
...

**Comment by Kor-Angar** [comment]
> игрок загружает сохраненку перед драконом и открывается пустой диалог с ним?

@dosinabox  есть сейв про такое?

**Reply by dosinabox** [reply]
Вот так быть НЕ должно:
https://youtu.be/XAfOSbS3DuY

[Saves_DragonTalk.zip](https://github.com/dosinabox/g2nr_unofficial_update/files/10834740/Saves_DragonTalk.zip)

**Reply by Kor-Angar** [reply]
это он к тебе обращается или ты к нему?

**Reply by Kor-Angar** [reply]
сейв от голд версии или неоф. обновления?

**Reply by dosinabox** [reply]
Непонятно кто к кому. Подозреваю, что без B_AssignDragonTalk такого не будет.
Сохранения из обновления г2нв.

**Reply by Kor-Angar** [reply]
у меня такого бага не получалось. в игре. возможно правки `B_AssignDragonTalk `  мне помогли.

но я бы добавил "выход" на всякий случай в ZS_Talk 
```D
FUNC void ZS_Talk ()
{
	// --- Keine Wahrnehmungen angemeldet ---
    Hlp_PrintConsole(Str_Format("       ZS_Talk [%s]",self.name));
    // EXIT IF...
    if (self.guild == GIL_DRAGON)
    {
        if (self.flags == 0)
        {
            return;
        };
    };
...
```

**Reply by dosinabox** [reply]
> у меня такого бага не получалось. в игре. возможно правки `B_AssignDragonTalk ` мне помогли.

а мне не помогли

**Reply by dosinabox** [reply]
хотя надо будет проверить со скомпилированными скриптами, без инъекций

**Reply by Kor-Angar** [reply]
попробуй добавить    `npcType = npctype_main;`

**Comment by Kor-Angar** [comment]
это работает
```D
func void B_AssignDragonTalk(var C_Npc slf)
{
};
/////////////////////////////////////////

instance DIA_Dragon_Fire_Exit(C_Info)
{
	DIA_Dragon_Fire_Exit_Old();
	npc = Dragon_Fire;
};

instance DIA_Dragon_Fire_Hello(C_Info)
{
	DIA_Dragon_Fire_Hello_Old();
	npc = Dragon_Fire;
};

instance DIA_Dragon_Fire_WerBistDu(C_Info)
{
	DIA_Dragon_Fire_WerBistDu_Old();
	npc = Dragon_Fire;
};

instance DIA_Dragon_Fire_HORT(C_Info)
{
	DIA_Dragon_Fire_HORT_Old();
	npc = Dragon_Fire;
};
```

**Reply by Kor-Angar** [reply]
```D
    func void DIA_Dragon_Fire_Hello_Info()
    {
        DIA_Dragon_Fire_Hello_Info_Old();
        B_AssignDragonTalk_Main(self);
    };
```

**Comment by Kor-Angar** [comment]
если герой сам инициировал конец разговора то было бы хорошо есть бы он что нибудь говорил дракону
```D
func void DIA_Dragon_Fire_Exit_Info()
{
    if (Npc_KnowsInfo(hero,DIA_Dragon_Fire_HORT) == false)
    || (Npc_KnowsInfo(hero,DIA_Dragon_Fire_WerBistDu) == false)
    {
        //Хватит болтать. Я загоню тебя обратно под камень, из-под которого ты вылез, монстр.    
        AI_Output(other,self,"DIA_Dragon_Undead_Hello_attack_15_00");  
    };	
    DIA_Dragon_Fire_Exit_Info_old();
};
```

**Reply by Kor-Angar** [reply]
например реплик бладвина подойдёт

AI_Output(other,self,"DIA_Addon_Bloodwyn_Wait_Raven_FIGHT_15_00"); //Хватит трепаться. Дерись.
AI_Output(self,other,"DIA_Dragon_Fire_Exit_20_00");	//Глаз потерял свою силу, и тебе не дожить до следующей зари.

**Comment by Kor-Angar** [comment]
А блок через зс_тэлк? 

```D
FUNC void ZS_Talk ()
{
	// --- Keine Wahrnehmungen angemeldet ---
    Hlp_PrintConsole(Str_Format("       ZS_Talk [%s]",self.name));
    // EXIT IF...
    if (self.guild == GIL_DRAGON)
    {
// наверное проверка на то заряжен ли амулет
        if (self.flags == 0)
        {
            return;
        };
    };

```
